microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc

```diff
@@ -0,0 +1,88 @@
+
+Most of the work has been done in the related issues below, so this RFE is more about testing and documenting.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-15016[WFLY-15016]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-14932[WFLY-14932 - Upgrade to MicroProfile Reactive Messaging 2.0] - This version introduces the new `@Channel` annotation, which makes it easier for users to get data into/from the Reactive Messaging Streams
+* https://issues.redhat.com/browse/WFLY-14932[WFLY-14932 - Configuration of Reactive Messaging Kafka messages] - The API introduced here allows you to get information about how Kafka handled a message, and to influence how messages are sent to Kafka. Most importantly it enables the use of a key for the messages sent to Kafka, which in turn inside Kafka are stored as records. Being able to query them, and not simply use Kafka as a messaging system is important.
+* https://issues.redhat.com/browse/WFLY-14987[WFLY-14987 - Ability to use an Elytron SSL Context in the Kafka connector] - The Kafka instance set up by AMQ Streams might be secured using SSL with unsigned certificates, so we need a way to specify the truststore.
+* https://issues.redhat.com/browse/WFLY-13640[WFLY-13640 - Provide Support for MicroProfile Reactive Messaging 1.0 for Kafka] - This was the original issue bringing Reactive Messaging 1.0 into WildFly. Some testing was done against AMQ Streams on OpenShift.
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [x] QE
+
+=== Affected Projects or Components
+WildFly
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+* It should be possible to connect to an AMQ Streams Kafka broker from an application.
+* The following security protocols are supported
+** `PLAINTEXT` - Communication is not encrypted and no autentication is needed
+* `SASL_PLAINTEXT` - No encryption but with SASL-based authentication
+* `TLS` - Communication is encrypted via either a CA or unsigned certificate. AMQ Streams supports the use of client certificates for optional authentication with this mechanism
+* `SASL_SSL` - Encrypted communication with SASL-based authentication
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+TBD as we identify pain points during testing. I am not intimately familiar with Kafka/AMQ Streams.
```

Comment:
Any updates to resolve this section? Anything we should mention? BTW I see there's an open bug affecting the implementation, i.e. https://issues.redhat.com/browse/WFWIP-367

Reply:
This one is out of our hands really, as mentioned in https://issues.redhat.com/browse/WFWIP-367?focusedId=15583303&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-15583303